builder/builder-next/builder.go

```diff
@@ -319,6 +320,17 @@ func (b *Builder) Build(ctx context.Context, opt backend.BuildConfig) (*builder.
 	}
 	frontendAttrs["add-hosts"] = extraHosts
 
+	if opt.Options.ShmSize > 0 {
+		frontendAttrs["shm-size"] = strconv.FormatInt(opt.Options.ShmSize, 10)
+	}
+
+	ulimits, err := toBuildkitUlimits(opt.Options.Ulimits)
+	if err != nil {
+		return nil, err
```

Comment:
These are also the ones we need to start looking at to return a correct error-type (InvalidParam()); per my other comment; there's a lot of those already, so let's make an effort to clean that up in some sweeps.